lib/braintrust/trace/contrib/github.com/alexrudall/ruby-openai/ruby-openai.rb

```diff
@@ -28,6 +28,85 @@ def self.parse_usage_tokens(usage)
+              aggregated = {
+                "id" => nil,
+                "created" => nil,
+                "model" => nil,
+                "usage" => nil,
+                "choices" => []
+              }
+
+              # Track aggregated content for the first choice
+              role = nil
+              content = ""
+
+              chunks.each do |chunk|
+                # Capture top-level fields from any chunk that has them
+                aggregated["id"] ||= chunk["id"]
+                aggregated["created"] ||= chunk["created"]
+                aggregated["model"] ||= chunk["model"]
+
+                # Aggregate usage (usually only in last chunk if stream_options.include_usage is set)
+                aggregated["usage"] = chunk["usage"] if chunk["usage"]
+
+                # Aggregate content from first choice
+                if chunk.dig("choices", 0, "delta", "role")
+                  role ||= chunk.dig("choices", 0, "delta", "role")
+                end
+                if chunk.dig("choices", 0, "delta", "content")
+                  content += chunk.dig("choices", 0, "delta", "content")
+                end
+              end
+
+              # Build aggregated choices array
+              aggregated["choices"] = [
+                {
+                  "index" => 0,
+                  "message" => {
+                    "role" => role || "assistant",
+                    "content" => content
+                  },
+                  "finish_reason" => chunks.dig(-1, "choices", 0, "finish_reason")
+                }
+              ]
+
+              aggregated
+            end
+
+            # Set span attributes from response data (works for both streaming and non-streaming)
+            # @param span [OpenTelemetry::Trace::Span] the span to set attributes on
+            # @param response_data [Hash] response hash with keys: choices, usage, id, created, model, system_fingerprint, service_tier
+            # @param time_to_first_token [Float] time to first token in seconds
+            # @param metadata [Hash] metadata hash to update with response fields
+            def self.set_span_attributes(span, response_data, time_to_first_token, metadata)
+              # Set output (choices) as JSON
+              if response_data["choices"]&.any?
+                span.set_attribute("braintrust.output_json", JSON.generate(response_data["choices"]))
+              end
+
+              # Set metrics (token usage + time_to_first_token)
+              metrics = {}
+              if response_data["usage"]
+                metrics = parse_usage_tokens(response_data["usage"])
+              end
+              metrics["time_to_first_token"] = time_to_first_token || 0.0
+              span.set_attribute("braintrust.metrics", JSON.generate(metrics)) unless metrics.empty?
+
+              # Update metadata with response fields
+              metadata["id"] = response_data["id"] if response_data["id"]
```

Comment:
i feel like this could be done in a loop more cleanly

Reply:
updated to loop over a list of fields instead